zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -767,9 +769,17 @@ private InterpreterSetting inlineCreateNewSetting(String name, String group,
                                                     InterpreterOption option,
                                                     Map<String, InterpreterProperty> properties)
       throws IOException {
-    if (name.indexOf(".") >= 0) {
-      throw new IOException("'.' is invalid for InterpreterSetting name.");
+    if (name == null) {
+      throw new IOException("Interpreter name shouldn't be empty.");
     }
+
+    // check if name is valid
+    Pattern pattern = Pattern.compile("^[-_a-zA-Z0-9]+$");
+    Matcher matcher = pattern.matcher(name);
+    if(!matcher.find()){
+      throw new IOException("Interpreter name shouldn't be empty, and can consist only of: -_a-zA-Z0-9");
```

Comment:
I'm not 100% sure that we should use `IOException` here, it's more like "Invalid parameters, or something like". But maybe it will require big changes along the call chain to use something else, so I'm ok with it...

Reply:
Yes, lazy me, I looked up what other APIs are using, and have used the same.

Reply:
I have a TODO list to go through all places & fix this :-)

Reply:
Regarding this, we do need to revisit that, and could do it in other PR